deployment/7_deploy_2.4.js

```diff
@@ -0,0 +1,345 @@
+
+const BACKWARD_COMPATIBILITY = 3;
+
+const main = async () => {
+  const { network, deploymentAccount, configurator, versionUploader, abiUploader } = await deployManager.getProps();
+
+  if (!["kovan", "kovan-fork", "staging", "prod"].includes(network)) {
+    console.warn("------------------------------------------------------------------------");
+    console.warn(`WARNING: The MakerManagerV2 module is not fully functional on ${network}`);
+    console.warn("------------------------------------------------------------------------");
+  }
+
+  const newModuleWrappers = [];
+  const newVersion = {};
+
+  // //////////////////////////////////
+  // Setup
+  // //////////////////////////////////
+
+  // if (1 + 1) return;
+  const { config } = configurator;
+
+  const ModuleRegistryWrapper = await ModuleRegistry.at(config.contracts.ModuleRegistry);
+  const MultiSigWrapper = await MultiSig.at(config.contracts.MultiSigWallet);
+  const multisigExecutor = new MultisigExecutor(MultiSigWrapper, deploymentAccount, config.multisig.autosign);
+
+  // //////////////////////////////////
+  // Deploy VersionManager module
+  // //////////////////////////////////
+
+  console.log("Deploying VersionManager module");
+  const VersionManagerWrapper = await VersionManager.new(
+    config.contracts.ModuleRegistry,
+    config.modules.LockStorage,
+    config.modules.GuardianStorage,
+    config.modules.TransferStorage,
+    config.modules.LimitStorage,
+  );
+  newModuleWrappers.push(VersionManagerWrapper);
+
+  // //////////////////////////////////
+  // Deploy features
+  // //////////////////////////////////
+
+  console.log("Deploying features");
+
+  // Deploy the GuardianManager module
+  const GuardianManagerWrapper = await GuardianManager.new(
+    config.modules.LockStorage,
+    config.modules.GuardianStorage,
+    VersionManagerWrapper.address,
+    config.settings.securityPeriod || 0,
+    config.settings.securityWindow || 0,
+  );
+  // Deploy the LockManager module
+  const LockManagerWrapper = await LockManager.new(
+    config.modules.LockStorage,
+    config.modules.GuardianStorage,
+    VersionManagerWrapper.address,
+    config.settings.lockPeriod || 0,
+  );
+  // Deploy the RecoveryManager module
+  const RecoveryManagerWrapper = await RecoveryManager.new(
+    config.modules.LockStorage,
+    config.modules.GuardianStorage,
+    VersionManagerWrapper.address,
+    config.settings.recoveryPeriod || 0,
+    config.settings.lockPeriod || 0,
+  );
+  // Deploy the ApprovedTransfer module
+  const ApprovedTransferWrapper = await ApprovedTransfer.new(
+    config.modules.LockStorage,
+    config.modules.GuardianStorage,
+    config.modules.LimitStorage,
+    VersionManagerWrapper.address,
+    config.defi.weth,
+  );
+  // Deploy the TransferManager module
+  const TransferManagerWrapper = await TransferManager.new(
+    config.modules.LockStorage,
+    config.modules.TransferStorage,
+    config.modules.LimitStorage,
```

Comment:
I'm a bit confused with what will happen to the daily limit of existing wallets with these settings. I think we should update the `init` method of the `TransferManager` to no longer migrate existing limits and save some gas.

Could be part of a separate PR if needed.

Reply:
For future reference, we agreed to change `TransferManager.init()` to only setup the default limit for new wallets. Recently updated wallets will have their limits stored in `LimitStorage` and we will no longer migrate the limits of older wallets that had their limits directly stored in the old `TransferManager`.